File: 0268-missing-number/0268-missing-number.py
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        n = len(nums)
        has = {}

        for i in range(n):
            has[nums[i]]=True
        
        for i in range(n+1):
            if i  not in has:
                return i
        return -1
        

File: 0268-missing-number/0268-missing-number.java
import java.util.*;

class Solution {
    public int missingNumber(int[] nums) {
        HashMap<Integer, Boolean> map = new HashMap<>();

        for(int i=0; i<nums.length; i++){
            map.put(nums[i], true);
        }


        for(int i=0; i<=nums.length; i++){
            if (map.containsKey(i)==false){
                return i;
            }
        }
        return 0;
    }
}